app/src/main/java/com/quran/labs/androidquran/ui/preference/QuranHeaderPreference.kt

```diff
@@ -0,0 +1,45 @@
+package com.quran.labs.androidquran.ui.preference
+
+import android.content.Context
+import android.graphics.Color
+import android.util.AttributeSet
+import android.widget.TextView
+import androidx.preference.Preference
+import androidx.preference.PreferenceViewHolder
+import com.quran.labs.androidquran.R
+
+class QuranHeaderPreference : Preference {
+
+  constructor(context: Context) : super(context) {
+    init()
+  }
+
+  constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
+    init()
+  }
+
+  constructor(
+    context: Context, attrs: AttributeSet, defStyleAttr: Int
+  ) : super(context, attrs, defStyleAttr) {
+    init()
+  }
+
+  constructor(
+    context: Context, attrs: AttributeSet, defStyleAttr: Int, defStyleRes: Int
+  ) : super(context, attrs, defStyleAttr, defStyleRes) {
+    init()
+  }
```

Comment:
how about something like this instead:

```kotlin
class QuranHeaderPreference @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0) : Preference(context, attires, defStyleAttr) {

   init {
      init()
   }
}
```

Reply:
what about the constructor with 4 params? It isn't used, should we remove it?

Reply:
true, please feel free to add defStyleRes: Int = 0 in the above and pass it to the `Preference` constructor to add it.

Reply:
Change applied, i also moved code inside init function to the init block